### ISSUE TITLE
Doesn't compile on 4.06.0+flambda: Fatal error: Assignment of a float to a specialised non-float array

### ISSUE THREAD

**Issue by copy** [janestreet/core_kernel#70]
On a clean 4.06.0+flamda switch:

```
#=== ERROR while installing core_kernel.v0.9.1 ================================#
# opam-version 1.2.2
# os           linux
# command      jbuilder build -p core_kernel -j 4
# path         /home/fabian/.opam/4.06.0-test-core-kernel/build/core_kernel.v0.9.1
# compiler     4.06.0+flambda
# exit-code    1
# env-file     /home/fabian/.opam/4.06.0-test-core-kernel/build/core_kernel.v0.9.1/core_kernel-14964-9f6e76.env
# stdout-file  /home/fabian/.opam/4.06.0-test-core-kernel/build/core_kernel.v0.9.1/core_kernel-14964-9f6e76.out
# stderr-file  /home/fabian/.opam/4.06.0-test-core-kernel/build/core_kernel.v0.9.1/core_kernel-14964-9f6e76.err
### stderr ###
# Unix.map_file raises Unix_error.
# [...]
#     ocamlopt src/core_kernel__Bigsubstring.{cmx,o}
#     ocamlopt src/core_kernel__Option_array.{cmx,o} (exit 2)
# (cd _build/default && /home/fabian/.opam/4.06.0-test-core-kernel/bin/ocamlopt.opt -w -40 -unsafe-string -g -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/base -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/base/caml -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/base/shadow_stdlib -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/bin_prot -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/bin_prot/shape -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/fieldslib -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/jane-street-headers -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/num -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/ocaml -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/ppx_assert/runtime-lib -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/ppx_bench/runtime-lib -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/ppx_compare/runtime-lib -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/ppx_expect/collector -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/ppx_expect/common -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/ppx_expect/config -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/ppx_hash/runtime-lib -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/ppx_inline_test/config -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/ppx_inline_test/runtime-lib -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/sexplib -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/sexplib/0 -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/stdio -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/typerep -I /home/fabian/.opam/4.06.0-test-core-kernel/lib/variantslib -I base_for_tests/src -no-alias-deps -I src -open Core_kernel__ -o src/core_kernel__Option_array.cmx -c -impl src/option_array.pp.ml)
# >> Fatal error: Assignment of a float to a specialised non-float array: (array.unsafe_set[int]<{src/option_array.ml:207,14-43;src/option_array.ml:145,21-79;src/uniform_array.ml:50,6-28;src/uniform_array.ml:33,27-66;src/obj_array.ml:79,4-74}>
#                                                           res/5490 i/5494
#                                                           obj/5504)
# Fatal error: exception Misc.Fatal_error
# 
# Waiting for 3 jobs to finish.
```

**Comment by hhugo** [comment]
This should be fixed with the upcoming release. cc @trefis
Meanwhile,  it should work with the development version (see https://github.com/janestreet/opam-repository)

**Comment by copy** [comment]
This has been fixed in v0.10.0.
